### PR TITLE
Upgrade to Elasticsearch 6.6.0.0

### PR DESCRIPTION
**Pull request by vvanholl**

### CODE REVIEW

**Comment by lukas-vlcek** [comment]
Hey there, I will have a chance to review this on Monday, I am on vacation atm.

File: src/main/java/org/elasticsearch/rest/prometheus/RestPrometheusMetricsAction.java [hunk 1]
```diff
@@ -36,10 +36,12 @@
  * REST action class for Prometheus Exporter plugin.
  */
 public class RestPrometheusMetricsAction extends BaseRestHandler {
+    Settings settings;
```
Comment (lukas-vlcek) [inline src/main/java/org/elasticsearch/rest/prometheus/RestPrometheusMetricsAction.java]: I would suggest using:
```java
private final Settings settings;
```

File: src/main/java/org/elasticsearch/action/NodePrometheusMetricsAction.java [hunk 2]
```diff
@@ -31,6 +31,7 @@ private NodePrometheusMetricsAction() {
     }
 
     @Override
+    @SuppressWarnings("deprecation")
```
Comment (lukas-vlcek) [inline src/main/java/org/elasticsearch/action/NodePrometheusMetricsAction.java]: I am not sure about this, we should rather follow recommendation from the code found [here](https://github.com/elastic/elasticsearch/commit/bf5d9a51c9360ce3c71998a6335c8e1be518a20a#diff-a10821429d1024280e86979d27f3734cR49).
It reads:
> Implement {`@`link #getResponseReader()} instead and make this method throw an {`@`link UnsupportedOperationException}

Comment (lukas-vlcek) [reply src/main/java/org/elasticsearch/action/NodePrometheusMetricsAction.java]: We need to keep the annotation for now. See comments in my commit.

File: src/main/java/org/elasticsearch/rest/prometheus/RestPrometheusMetricsAction.java (hunk 1 above)
Comment (lukas-vlcek) [reply src/main/java/org/elasticsearch/rest/prometheus/RestPrometheusMetricsAction.java]: Pushed new commit with needed changes.

**Comment by lukas-vlcek** [comment]
@vvanholl Please feel free to review my updates to this. I can see that currently the merging is blocked due to: "The base branch requires all commits to be signed." So let me know if you need me to sign the commit and re-push into branch before merging.

**Comment by rajugupta15** [comment]
When can i expect this to be merge?

**Comment by nikolay** [comment]
@vvanholl @lukas-vlcek By the way, meanwhile, v6.6.1 got released as well.

**Comment by lukas-vlcek** [comment]
I can not merge it and prepare distribution release myself. We need to wait for @vvanholl.

**Comment by lukas-vlcek** [comment]
@nikolay Would you mind opening a ticket to release the plugin for ES `v6.6.1` ?

**Comment by vvanholl** [comment]
Let's merge and publish now

**Comment by nikolay** [comment]
@lukas-vlcek I will.

**Pull request merged**